pkg/ingress/endpoint.go

```diff
@@ -114,6 +114,21 @@ func (c *endpointsController) sync(ctx context.Context, ev *types.Event) error {
 		portMap[port.Name] = port.Port
 	}
 	clusters := c.controller.apisix.ListClusters()
+
+	//endpoints turn to zero
+	if len(ep.Subsets) == 0 {
+		nodes := []apisixv1.UpstreamNode{}
```

Comment:
We may still need to respect the ApisixUpstream subset settings, which means we need to traverse the ApisixUpstream subsets and reset the nodes to empty for all of them.